libraries/chain/include/eosio/chain/finality/finality_core.hpp

```diff
@@ -33,6 +33,7 @@ struct qc_claim_t
    block_num_type  block_num {0};
    bool            is_strong_qc {false};
 
+   // [greg] I think we should use only `block_num` in the comparison
```

Comment:
I don't think so. We want a strong QC claim on a block to be able to override a weak QC claim on the same block. This could be possible if after putting the weak QC in a block, a few extra strong votes arrive on a pending QC of a block; then the next block has the opportunity to provide the strong QC assuming it doesn't have an even better QC available by then.

Reply:
Thanks, seems right, I'm not sure what I was thinking.